### PR TITLE
Add new theme: base16-256

### PR DESCRIPTION
**Pull request by mk12**
See #1033 for the discussion that led to this.

In summary, the base16 theme introduced in #543 works great for users who configure their terminals with the [base16 system](https://github.com/chriskempson/base16) in which the 16 ANSI colors are repurposed as base16 palette slots. But it doesn't work for users of [base16-shell](https://github.com/chriskempson/base16-shell), which aims to solve problems with base16-unaware software by leaving the bright colors (ANSI 8-15) alone, and instead using codes 16-21 for extra base16 palette slots (requiring a 256-color terminal, hence base16-256).

Here is how terminal themes for [regular base16](https://github.com/kdrag0n/base16-kitty/blob/master/templates/default.mustache) and [256-color/base16-shell](https://github.com/kdrag0n/base16-kitty/blob/master/templates/default-256.mustache) differ:

| ANSI code | base16 | base16-256 |
|---|---|---|
| 08 | base03 | base03 |
| 09 | **base09** | base08 |
| 10 | **base01** | base0B |
| 11 | **base02** | base0A |
| 12 | **base04** | base0D |
| 13 | **base06** | base0E |
| 14 | **base0F** | base0C |
| 15 | base07 | base07 |
| 16 | | **base09** |
| 17 | | **base0F** |
| 18 | | **base01** |
| 19 | | **base02** |
| 20 | | **base04** |
| 21 | | **base06** |

Notice codes 08 (bright black) and 15 (bright white) stay the same. But 09 (bright red) through 14 (bright cyan) are replaced with their non-bright counterparts, and those base16 colors are moved to ANSI codes 16-21.

So the new base16-256 theme is just like base16, but makes the following replacements:

- base09 goes from ANSI 09 to 16: `#09000000` → `#10000000`
- base01 goes from ANSI 10 to 18: `#0a000000` → `#12000000`
- base02 goes from ANSI 11 to 19: `#0b000000` → `#13000000`
- base04 goes from ANSI 12 to 20: `#0c000000` → `#14000000`
- base06 goes from ANSI 13 to 21: `#0d000000` → `#15000000`
- base0F goes from ANSI 14 to 17: `#0e000000` → `#11000000`

terminal.rs already maps any `#RR000000` to the 256-color code `RR`, so no code change is needed.

Here is what base16 and base16-256 look like on macOS, kitty terminal with [https://github.com/kdrag0n/base16-kitty/blob/master/colors/base16-solarized-light-256.conf](base16-solarized-light-256.conf), and fish shell with [base16-shell/profile_helper.fish](https://github.com/chriskempson/base16-shell/blob/master/profile_helper.fish) sourced:

Screen Shot 2020-07-19 at 5 22 08 PM

It's not so obvious in this example, but in this theme constants like `2.0` are supposed to be orange, not red. The base16 theme shows red because it's using ANSI bright red (which is just plain red in the 256-color terminal theme). The base16-256 theme correctly shows it as orange because it instead uses ANSI code 16.

Note: I have not regenerated themes.bin — I assume that will be done on release as before.

### CODE REVIEW

**Comment by antoineco** [comment]
Looks much better in 256-color terminal themes 👍 

YAML example, to refer to the linked issue

_before:_
image

_after:_
image

File: assets/themes/base16.tmTheme [hunk 1]
```diff
@@ -3,14 +3,21 @@
...
-        palette number from 00 to 0f, and AA is the special value 00 to indicate
-        that this encoding is being used.
+        palette number from 00 to 0f (the 8 colors and their bright versions),
+        and AA is the special value 00 to indicate this encoding is being used.
         -->
         <key>author</key>
         <string>Template: Chris Kempson, Scheme: Mitchell Kember</string>
         <key>name</key>
-        <string>Base16</string>
+        <string>Base16 256</string>
```
Comment (antoineco) [inline assets/themes/base16.tmTheme:20]: Shouldn't this one remain `Base16` and the other one be named `Base16 256`?

Comment (mk12) [reply assets/themes/base16.tmTheme]: Yes, thanks for catching that. Fixed.

**Comment by mk12** [comment]
@sharkdp Please take a look when you have time :)

I've also updated the README with a section on the 8-bit color themes (ansi-dark, ansi-light, base16, and base16-256) to help avoid confusion about them.

**Comment by sharkdp** [comment]
Thank you very much for your contribution and for the awesome documentation (and updating the README)!

I have only one minor request: Could you please add an entry to the "unreleased" section in the `CHANGELOG.md`? The format is:
```
- Description what has been changed, see #123 (@user)
```
where `#123` links to the bug ticket and/or this PR and `user` is your username.

**Comment by sharkdp** [comment]
I rebased your branch on master in order to run the new CI pipelines. I am expecting one (new) test to fail, which regenerates the binary caches before running the unit tests. We simply have to add the theme to `tests/assets.rs` to make it pass.

**Comment by mk12** [comment]
> I have only one minor request: Could you please add an entry to the "unreleased" section in the CHANGELOG.md?

Done! (I amended the 1st commit and force pushed)

**Comment by sharkdp** [comment]
Thank you for the update.

My new testing mechanism catches the (future) error, but now the tests for the old binary theme set fail, of course. I'm adding the updated `themes.bin` to this branch for now and will further work on my test suite :smile:

**Comment by sharkdp** [comment]
Added in [v0.16.0](https://github.com/sharkdp/bat/releases/tag/v0.16.0).

**Pull request merged**